File: neutralino/resources/js/0types.ts
export type tIdAndDate = { id: string; date: Date }

export type tTaskGroup = {
  id: string;
  version: number;
  tasks: tTask[];
}

export type tTask = {
  id: string;
  name: string;
  days: number;
  isArchived: boolean;
  datesDone: tIdAndDate[];
}

export type tConfig = {
  dataFilePath: string
  shouldNotify: boolean;
  shouldMinimizeToTrayOnQuit: boolean
}

export type tStore = {
  taskGroups: tTaskGroup[];
  config: tConfig;
}

File: fyne/src/0types/0types.go
package types

import "time"

type IdAndDate struct {
	Id   string
	Date time.Time
}

type Task struct {
	Id         string
	Name       string
	Days       int
	IsArchived bool
	DatesDone  []IdAndDate
}

type TaskGroup struct {
	Id      string
	Version int
	Tasks   []Task
}

type Config struct {
	DataFilePath               string
	ConfigFilePath             string
	ShouldNotify               bool
	ShouldMinimizeToTrayOnQuit bool
}

type Store struct {
	TaskGroups []TaskGroup
	Config     Config
}
